File: 61/solve.py
def solve(base, exp):
    assert exp >= 0

    pows = [(0, 1), (1, base)]
    current_pow = 1
    while current_pow < exp:
        current_pow *= 2
        pows.append((current_pow, pows[-1][1] * pows[-1][1]))

    result = 1
    index = len(pows) - 1
    while exp > 0:
        if exp >= pows[index][0]:
            exp -= pows[index][0]
            result *= pows[index][1]
        else:
            index -= 1

    return result

File: The Josephus Problem/solve.go
package main

func solve(n int) int {
	if n == 2 {
		return 1
	}

	if n%2 == 0 {
		return 2*solve(n/2) - 1
	}

	return f(n, solve(n/2+1))
}

func f(n int, pos int) int {
	if pos == 1 {
		return n
	}
	return 2*pos - 1
}
